packages/components/src/color-picker/picker.tsx

```diff
@@ -7,18 +7,71 @@ import { colord } from 'colord';
 /**
  * WordPress dependencies
  */
-import { useMemo } from '@wordpress/element';
+import { useMemo, useEffect, useRef } from '@wordpress/element';
 /**
  * Internal dependencies
  */
 import type { PickerProps } from './types';
 
-export const Picker = ( { color, enableAlpha, onChange }: PickerProps ) => {
+export const Picker = ( {
+	color,
+	enableAlpha,
+	onChange,
+	onDragStart,
+	onDragEnd,
+}: PickerProps ) => {
 	const Component = enableAlpha
 		? RgbaStringColorPicker
 		: RgbStringColorPicker;
 	const rgbColor = useMemo( () => color.toRgbString(), [ color ] );
 
+	const isDragging = useRef( false );
+	const leftWhileDragging = useRef( false );
+	useEffect( () => {
+		const picker = document.querySelector( '.react-colorful__saturation' );
```

Comment:
Is it safe to use `document.querySelector` here? I wasn't too sure, but this stood out to me as something that's potentially doing a query that could be outside of the scope of this particular component. Would it be worth / or possible to grab a `ref` for this component, and then do a `.querySelector` on the component's node instead to make sure we're only looking at children of the current component?

Apologies if I'm missing details here!

Reply:
> Is it safe to use `document.querySelector` here?

Probably not 😅  I implemented your suggestion in https://github.com/WordPress/gutenberg/pull/55149/commits/74e1418ca9df91d013197d241fb99e2acecbf2c7,  better safe than sorry!